publish-wrapper

```diff
@@ -0,0 +1,204 @@
+                            }
+                        ]
+                    }
+                }]
+            },
+            "revision": opts.revision,
+            "onaborted": {
+                "github": {
+                    "token": self.api.token,
+                    "requests": [
+                        # Set status to error
+                        {
+                            "method": "POST",
+                            "resource": self.api.qualify("statuses/" + opts.revision),
+                            "data": {
+                                "state": "error",
+                                "context": opts.github_context,
+                                "description": "Aborted without status",
+                                "target_url": ":link"
+                            }
+                        }
+                    ]
+                },
+            }
+        }
+
+        status["link"] = "log.html"
+        status["extras"] = ["https://raw.githubusercontent.com/cockpit-project/bots/master/task/log.html"]
+
+        # For other scripts to use
+        os.environ["TEST_DESCRIPTION"] = description
+        self.sink = sink.Sink(opts.publish, identifier, status)
+
+    def stop_publishing(self, ret):
+        sink = self.sink
+
+        def mark_failed():
+            if "github" in sink.status:
+                self.github_status_data["state"] = "failure"
+
+        def mark_passed():
+            if "github" in sink.status:
+                self.github_status_data["state"] = "success"
+
+        if isinstance(ret, str):
+            message = ret
+            mark_failed()
+            ret = 0
+        elif ret == 0:
+            message = "Tests passed"
+            mark_passed()
+        else:
+            message = "Tests failed with code {0}".format(ret)
+            mark_failed()
+            ret = 0  # A failure, but not for this script
+        sink.status["message"] = message
+        if "github" in sink.status:
+            self.github_status_data["description"] = message
+        try:
+            del sink.status["extras"]
+        except KeyError:
+            pass
+        sink.flush()
+
+        return ret
+
+    def detect_collisions(self, opts):
+        statuses = self.api.get("commits/{0}/statuses".format(opts.revision))
+        for status in statuses:
+            if status.get("context") == opts.github_context:
+                if status.get("state") != "pending":
+                    return "Status is not pending"
+                if status.get("description") not in [github.NOT_TESTED, github.NOT_TESTED_DIRECT]:
+                    return "Status description is not in [{0}, {1}]".format(
+                        github.NOT_TESTED, github.NOT_TESTED_DIRECT)
+                return None  # only check the newest status of the supplied context
+        return None
+
+    def run(self, opts):
+        ret = self.detect_collisions(opts)
+        if ret:
+            sys.stderr.write(ret)
```

Comment:
The previous version prefixed the error with "Collision detected:". Did you remove that on purpose?